File: 백준/Silver/20003. 거스름돈이 싫어요/거스름돈이 싫어요.cc
#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define MOD 1000000007
#define MAX 100000
#define INF 987654321

ll gcd(ll a, ll b)
{
    if (b == 0)
        return a;
    return gcd(b, a % b);
}
int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    ll n;
    cin >> n;
    vector<pair<ll, ll>> v;
    for (ll i = 0; i < n; i++)
    {
        ll a, b;
        cin >> a >> b;
        ll temp = gcd(a, b);
        v.push_back({a / temp, b / temp});
    }

    ll demo = v[0].second;

    for (ll i = 1; i < n; i++)
        demo = demo * v[i].second / gcd(demo, v[i].second);

    ll nume = v[0].first * (demo / v[0].second);

    for (ll i = 1; i < n; i++)
        nume = gcd(nume, v[i].first * (demo / v[i].second));

    cout << nume << " " << demo;
    return 0;
}

File: 백준/Silver/20003. 거스름돈이 싫어요/거스름돈이 싫어요.py
def gcd(a, b):
    if b == 0:
        return a
    else:
        return gcd(b, a % b)


n = int(input())
v = []

for _ in range(n):
    a, b = map(int, input().split())
    c = gcd(a, b)
    v.append((a // c, b // c))
deno = v[0][1]

for i in range(1, n):
    deno = (deno * v[i][1]) // gcd(deno, v[i][1])

nume = v[0][0] * (deno // v[0][1])

for i in range(1, n):
    nume = gcd(nume, v[i][0] * (deno // v[i][1]))

temp = gcd(deno, nume)
print(nume // temp, deno // temp)
